demos/coreMQTT/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1906 @@
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://www.FreeRTOS.org
+ * http://aws.amazon.com/freertos
+ */
+
+/*
+ * Demo for showing use of the managed MQTT API shared between multiple tasks.
+ * This demo uses a thread safe queue to hold commands for interacting with the
+ * MQTT API. A command task processes commands from the queue while other tasks
+ * enqueue them. This task enters a loop, during which it processes commands from
+ * the command queue. If a termination command is received, it will break from
+ * the loop. In addition to the command task, this demo uses one task for
+ * publishing messages to the MQTT broker and another for receiving them via
+ * an MQTT subscription. The publisher task creates a series of publish operations
+ * to push to the command queue, which are then executed by the command task.
+ * The subscriber task subscribes to a topic filter matching the topics published
+ * on by the publisher, and then loops while waiting for publish messages to be
+ * received. Each task has a queue to hold received publish messages,
+ * and the command task pushes incoming publishes to the queue of each task
+ * that is subscribed to the incoming topic.
+ *
+ * !!! NOTE !!!
+ * This MQTT demo does not authenticate the server nor the client.
+ * Hence, this demo should not be used as production ready code.
+ */
+
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+#include <assert.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+
+/* Demo Specific configs. */
+#include "mqtt_demo_connection_sharing_config.h"
+
+/* MQTT library includes. */
+#include "core_mqtt.h"
+#include "core_mqtt_state.h"
+
+/* Retry utilities include. */
+#include "retry_utils.h"
+
+/* Transport interface implementation include header for TLS. */
+#include "transport_secure_sockets.h"
+
+/* Credentials for the MQTT broker endpoint. */
+#include "aws_clientcredential.h"
+
+/**
+ * These configuration settings are required to run the demo.
+ */
+#ifndef democonfigCLIENT_IDENTIFIER
+
+/**
+ * @brief The MQTT client identifier used in this example.  Each client identifier
+ * must be unique so edit as required to ensure no two clients connecting to the
+ * same broker use the same client identifier.
+ */
+    #define democonfigCLIENT_IDENTIFIER    clientcredentialIOT_THING_NAME
+#endif
+
+/* Compile time error for some undefined configs, and provide default values
+ * for others. */
+#ifndef democonfigMQTT_BROKER_ENDPOINT
+    #define democonfigMQTT_BROKER_ENDPOINT    clientcredentialMQTT_BROKER_ENDPOINT
+#endif
+
+#ifndef democonfigMQTT_BROKER_PORT
+    #define democonfigMQTT_BROKER_PORT    clientcredentialMQTT_BROKER_PORT
+#endif
+
```

Comment:
Can you please add this below. The CI will replace this macro in this file directly.
```suggestion

#ifndef democonfigROOT_CA_PEM
    #define democonfigROOT_CA_PEM    tlsATS1_ROOT_CERTIFICATE_PEM
#endif
```